volume/nfs/nfs.go

```diff
@@ -2,64 +2,72 @@ package nfsvolumedriver
 
 import (
 	"errors"
+	"fmt"
+	"github.com/Sirupsen/logrus"
+	"github.com/docker/docker/volume"
+	"github.com/opencontainers/runc/libcontainer"
+	"io/ioutil"
+	"os"
+	"os/exec"
 	"sync"
+)
 
-	"github.com/docker/docker/volume"
+const (
+	NFS_MOUNTS_FOLDER             = "/var/lib/docker/nfs_mounts"
+	NFS_MOUNTS_FOLDER_PERMISSIONS = 0755
 )
 
 func New() *Root {
-	return &Root{
-		volumes: make(map[string]*Volume),
-	}
+	return &Root{}
 }
 
 type Root struct {
-	m       sync.Mutex
-	volumes map[string]*Volume
+	m sync.Mutex
 }
 
 func (r *Root) Name() string {
 	return "nfs"
 }
 
+// Makes sure that the nfs mounts folder exists
+func ensureNfsFolderExists() error {
+	_, err := os.Stat(NFS_MOUNTS_FOLDER)
+	if err == nil {
+		return nil
+	}
+	if !os.IsNotExist(err) {
+		return err
+	}
+	return os.MkdirAll(NFS_MOUNTS_FOLDER, NFS_MOUNTS_FOLDER_PERMISSIONS)
+}
+
 func (r *Root) Create(name string, _ map[string]string) (volume.Volume, error) {
 	r.m.Lock()
 	defer r.m.Unlock()
 
-	v, exists := r.volumes[name]
-	if !exists {
-		v = &Volume{
-			driverName:       r.Name(),
-			name:             name,
-		}
-		r.volumes[name] = v
-	}
-	v.use()
-	return v, nil
+	ensureNfsFolderExists()
```

Comment:
No, wait. I remember why I placed it here. This way it's using the Root lock intsead of a particular volume lock. Otherwise, it might get called twice if two volumes are created and mounted at the same time.